install/kubernetes/istio-gateway.yaml

```diff
@@ -0,0 +1,55 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: istio-gateway
+spec:
+  selector:
+    istio: ingressgateway # use istio default controller
+  servers:
+  - port:
+      number: 15007
+      name: pilot-http-discovery
+      protocol: TCP
+    hosts:
+    - "*"
+  - port:
+      number: 15010
+      name: pilot-grpc-xds
+      protocol: TCP
+    hosts:
+    - "*"
```

Comment:
indeed we don't need all ports.  @gyliu513 had a list of ports he thought pilot needed - I think they match up with what you have here.  @gyliu513 can you comment?

Reply:
I'm also not clear which port should be exposed as well, I was checking the remote cluster, and found that only port 15007 was used for pilot if deployed without mTLS, so for remote cluster, we may only need 15007 for pilot.

```
- args:
        - proxy
        - sidecar
        - --configPath
        - /etc/istio/proxy
        - --binaryPath
        - /usr/local/bin/envoy
        - --serviceCluster
        - details
        - --drainDuration
        - 45s
        - --parentShutdownDuration
        - 1m0s
        - --discoveryAddress
        - istio-pilot.istio-system:15007
        - --discoveryRefreshDelay
        - 10s
        - --zipkinAddress
        - zipkin.istio-system:9411
```

But for this new solution, I think we can start from 15007 first and do some test first.

Reply:
Unfortunately it is dangerous to not expose all the ports as some of the ports are probably used in configurations or with options we didn't try.  IMO we should take the opposite approach err on the side of exposing more ports than might be needed and let experts tell us which ports can be removed.

Reply:
@gyliu513 grpc uses port 15010. I'm not sure what the argument discoveryAddress is doing right now. It might need a little bit clean up for it

Reply:
15007 is used for v1.